src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -482,14 +526,15 @@ private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
         private async Task<T?> PerformNuGetProjectServiceWorkAsync<T>(
             Func<INuGetProjectService, CancellationToken, ValueTask<T?>> doWorkAsync, CancellationToken cancellationToken)
         {
+            // Make sure we are on the thread pool to avoid UI thread dependencies if external code uses ConfigureAwait(true).
+            // GetServiceAsync/GetProxyAsync and the cast below are all explicitly documented as being BG thread safe.
+            await TaskScheduler.Default;
```

Comment:
📝 Would greatly prefer this context switch and comment be returned to its original location/form. The relocation doesn't improve performance of the code now under the umbrella of the switch, but it does move a required switch further away from the location where it is required.